datasets/loader.py

```diff
@@ -174,32 +358,257 @@ def klaverjas(dataset_dir=None):
+    gisette y train dataset (6000, 1)
+    gisette X test dataset  (1000, 5000)
+    gisette y test dataset  (1000, 1)
     """
-    dataset_name = 'connect'
+    dataset_name = 'gisette'
     os.makedirs(dataset_dir, exist_ok=True)
 
-    X, y = fetch_openml(name='connect-4', return_X_y=True,
+    cache_dir = os.path.join(dataset_dir, '_gisette')
+    os.makedirs(cache_dir, exist_ok=True)
+
+    domen_hhtp = 'http://archive.ics.uci.edu/ml/machine-learning-databases/'
+
+    gisette_train_data_url = domen_hhtp + '/gisette/GISETTE/gisette_train.data'
+    filename_train_data = os.path.join(cache_dir, 'gisette_train.data')
+    if not os.path.exists(filename_train_data):
+        _retrieve(gisette_train_data_url, filename_train_data)
+
+    gisette_train_labels_url = domen_hhtp + '/gisette/GISETTE/gisette_train.labels'
+    filename_train_labels = os.path.join(cache_dir, 'gisette_train.labels')
+    if not os.path.exists(filename_train_labels):
+        _retrieve(gisette_train_labels_url, filename_train_labels)
+
+    gisette_test_data_url = domen_hhtp + '/gisette/GISETTE/gisette_valid.data'
+    filename_test_data = os.path.join(cache_dir, 'gisette_valid.data')
+    if not os.path.exists(filename_test_data):
+        _retrieve(gisette_test_data_url, filename_test_data)
+
+    gisette_test_labels_url = domen_hhtp + '/gisette/gisette_valid.labels'
+    filename_test_labels = os.path.join(cache_dir, 'gisette_valid.labels')
+    if not os.path.exists(filename_test_labels):
+        _retrieve(gisette_test_labels_url, filename_test_labels)
+
+    logging.info('gisette dataset is downloaded')
+    logging.info('reading CSV file...')
+
+    num_cols = 5000
+
+    df_train = pd.read_csv(filename_train_data, header=None)
+    df_labels = pd.read_csv(filename_train_labels, header=None)
+    num_train = 6000
+    x_train_arr = df_train.iloc[:num_train].values
+    x_train = pd.DataFrame(np.array([np.fromstring(
+        elem[0], dtype=int, count=num_cols, sep=' ') for elem in x_train_arr]))
+    y_train_arr = df_labels.iloc[:num_train].values
+    y_train = pd.DataFrame((y_train_arr > 0).astype(int))
+
+    num_train = 1000
+    df_test = pd.read_csv(filename_test_data, header=None)
+    df_labels = pd.read_csv(filename_test_labels, header=None)
+    x_test_arr = df_test.iloc[:num_train].values
+    x_test = pd.DataFrame(np.array(
+        [np.fromstring(
+            elem[0],
+            dtype=int, count=num_cols, sep=' ')
+         for elem in x_test_arr]))
+    y_test_arr = df_labels.iloc[:num_train].values
+    y_test = pd.DataFrame((y_test_arr > 0).astype(int))
+
+    for data, name in zip((x_train, x_test, y_train, y_test),
+                          ('x_train', 'x_test', 'y_train', 'y_test')):
+        filename = f'{dataset_name}_{name}.csv'
+        data.to_csv(os.path.join(dataset_dir, filename),
+                    header=False, index=False)
+
+    logging.info('dataset gisette ready.')
+    return True
+
+
+def higgs(dataset_dir: Path) -> bool:

```

Comment:
What size?

Usually used:
10M train and 1M test
or 
10.5M train and 0.5M test

Reply:
I guess it's just the whole dataset. Will find out some info about it